The job **`mingw-check-tidy`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/13732903719/job/38412869749) [(plain)](https://github.com/rust-lang/rust/commit/90ebc246075f9b56cd0a2dee2496698a94c9c033/checks/38412869749/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
info: removing rustup binaries
info: rustup is uninstalled
##[group]Image checksum input
mingw-check-tidy
# We use the ghcr base image because ghcr doesn't have a rate limit
# and the mingw-check-tidy job doesn't cache docker images in CI.
FROM ghcr.io/rust-lang/ubuntu:22.04
ARG DEBIAN_FRONTEND=noninteractive
RUN apt-get update && apt-get install -y --no-install-recommends \
  g++ \
  make \
---

COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
COPY host-x86_64/mingw-check/validate-error-codes.sh /scripts/

# NOTE: intentionally uses python2 for x.py so we can test it still works.
# validate-toolstate only runs in our CI, so it's ok for it to only support python3.
ENV SCRIPT TIDY_PRINT_DIFF=1 python2.7 ../x.py test \
           --stage 0 src/tools/tidy tidyselftest --extra-checks=py,cpp
# This file is autogenerated by pip-compile with Python 3.10
# by the following command:
#
#    pip-compile --allow-unsafe --generate-hashes reuse-requirements.in
---
#12 3.080 Building wheels for collected packages: reuse
#12 3.081   Building wheel for reuse (pyproject.toml): started
#12 3.294   Building wheel for reuse (pyproject.toml): finished with status 'done'
#12 3.295   Created wheel for reuse: filename=reuse-4.0.3-cp310-cp310-manylinux_2_35_x86_64.whl size=132720 sha256=0c2fd2aaf7b0bf8d6e131220aff14712a774c2ca462f3204d25460cbcf610b63
#12 3.296   Stored in directory: /tmp/pip-ephem-wheel-cache-jt45esn3/wheels/3d/8d/0a/e0fc6aba4494b28a967ab5eaf951c121d9c677958714e34532
#12 3.298 Installing collected packages: boolean-py, binaryornot, tomlkit, reuse, python-debian, markupsafe, license-expression, jinja2, chardet, attrs
#12 3.700 Successfully installed attrs-23.2.0 binaryornot-0.4.4 boolean-py-4.0 chardet-5.2.0 jinja2-3.1.4 license-expression-30.3.0 markupsafe-2.1.5 python-debian-0.1.49 reuse-4.0.3 tomlkit-0.13.0
#12 3.700 WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
#12 4.230 Collecting virtualenv
#12 4.230 Collecting virtualenv
#12 4.318   Downloading virtualenv-20.29.3-py3-none-any.whl (4.3 MB)
#12 4.454      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 4.3/4.3 MB 32.1 MB/s eta 0:00:00
#12 4.494 Collecting distlib<1,>=0.3.7
#12 4.513   Downloading distlib-0.3.9-py2.py3-none-any.whl (468 kB)
#12 4.521      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 469.0/469.0 KB 82.2 MB/s eta 0:00:00
#12 4.555 Collecting platformdirs<5,>=3.9.1
#12 4.576   Downloading platformdirs-4.3.6-py3-none-any.whl (18 kB)
#12 4.611 Collecting filelock<4,>=3.12.2
#12 4.631   Downloading filelock-3.17.0-py3-none-any.whl (16 kB)
#12 4.710 Installing collected packages: distlib, platformdirs, filelock, virtualenv
#12 4.889 Successfully installed distlib-0.3.9 filelock-3.17.0 platformdirs-4.3.6 virtualenv-20.29.3
#12 DONE 5.0s

#13 [7/8] COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
#13 DONE 0.0s
---
DirectMap4k:      120768 kB
DirectMap2M:     7219200 kB
DirectMap1G:    11534336 kB
##[endgroup]
Executing TIDY_PRINT_DIFF=1 python2.7 ../x.py test            --stage 0 src/tools/tidy tidyselftest --extra-checks=py,cpp
+ TIDY_PRINT_DIFF=1 python2.7 ../x.py test --stage 0 src/tools/tidy tidyselftest --extra-checks=py,cpp
    Finished `dev` profile [unoptimized] target(s) in 0.05s
##[endgroup]
WARN: currently no CI rustc builds have rustc debug assertions enabled. Please either set `rust.debug-assertions` to `false` if you want to use download CI rustc or set `rust.download-rustc` to `false`.
downloading https://static.rust-lang.org/dist/2025-02-18/rustfmt-nightly-x86_64-unknown-linux-gnu.tar.xz
---
Diff in /checkout/src/tools/unicode-table-generator/src/skiplist.rs:68:
             assert!(inserted);
         }
 
+        writeln!(&mut self.file, "use super::ShortOffsetRunHeader;\n").unwrap();
             &mut self.file,
-            "use super::ShortOffsetRunHeader;\n"
-        ).unwrap();
-        writeln!(
-        writeln!(
-            &mut self.file,
             "static SHORT_OFFSET_RUNS: [ShortOffsetRunHeader; {}] = [{}];",
             fmt_list(short_offset_runs.iter())
Diff in /checkout/src/tools/unicode-table-generator/src/skiplist.rs:111:
         } else {
         } else {
             writeln!(&mut self.file, "pub fn lookup(c: char) -> bool {{").unwrap();
         }
+        writeln!(&mut self.file, "    const {{",).unwrap();
             &mut self.file,
-            "    const {{",
-        )
-        .unwrap();
-        .unwrap();
-        writeln!(
-            &mut self.file,
             "        assert!(SHORT_OFFSET_RUNS.last().unwrap().0 > (char::MAX as u32));",
         .unwrap();
Diff in /checkout/src/tools/unicode-table-generator/src/skiplist.rs:124:
Diff in /checkout/src/tools/unicode-table-generator/src/skiplist.rs:124:
+        writeln!(&mut self.file, "        let mut i = 0;",).unwrap();
+        writeln!(&mut self.file, "        while i < SHORT_OFFSET_RUNS.len() {{",).unwrap();
             &mut self.file,
-            "        let mut i = 0;",
-        )
-        .unwrap();
-        .unwrap();
-        writeln!(
-            &mut self.file,
-            "        while i < SHORT_OFFSET_RUNS.len() {{",
-        .unwrap();
-        writeln!(
-            &mut self.file,
-            &mut self.file,
             "            assert!(SHORT_OFFSET_RUNS[i].start_index() < OFFSETS.len());",
         .unwrap();
Diff in /checkout/src/tools/unicode-table-generator/src/skiplist.rs:139:
-        writeln!(
-            &mut self.file,
-            &mut self.file,
-            "            i += 1;",
-        .unwrap();
-        writeln!(
-            &mut self.file,
-            "        }}",
-            "        }}",
-        )
-        .unwrap();
-        writeln!(
-            &mut self.file,
-            "    }}",
-        )
-        .unwrap();
+        writeln!(&mut self.file, "            i += 1;",).unwrap();
+        writeln!(&mut self.file, "        }}",).unwrap();
+        writeln!(&mut self.file, "    }}",).unwrap();
             &mut self.file,
             &mut self.file,
             "    // SAFETY: We just ensured the last element of `SHORT_OFFSET_RUNS` is greater than `std::char::MAX`",
Build completed unsuccessfully in 0:01:13
  local time: Sat Mar  8 02:21:52 UTC 2025
  network time: Sat, 08 Mar 2025 02:21:52 GMT
##[error]Process completed with exit code 1.
```

</details>

--- a/library/core/src/unicode/unicode_data.rs
+++ b/library/core/src/unicode/unicode_data.rs
@@ -224,7 +224,7 @@ pub mod alphabetic {
     ];
     pub fn lookup(c: char) -> bool {
         const {
-            assert!(SHORT_OFFSET_RUNS.last().unwrap().0 > (char::MAX as u32));
+            assert!(SHORT_OFFSET_RUNS.last().unwrap().0 > char::MAX as u32);
             let mut i = 0;
             while i < SHORT_OFFSET_RUNS.len() {
                 assert!(SHORT_OFFSET_RUNS[i].start_index() < OFFSETS.len());
@@ -299,7 +299,7 @@ pub mod case_ignorable {
     ];
     pub fn lookup(c: char) -> bool {
         const {
-            assert!(SHORT_OFFSET_RUNS.last().unwrap().0 > (char::MAX as u32));
+            assert!(SHORT_OFFSET_RUNS.last().unwrap().0 > char::MAX as u32);
             let mut i = 0;
             while i < SHORT_OFFSET_RUNS.len() {
                 assert!(SHORT_OFFSET_RUNS[i].start_index() < OFFSETS.len());
@@ -345,7 +345,7 @@ pub mod cased {
     ];
     pub fn lookup(c: char) -> bool {
         const {
-            assert!(SHORT_OFFSET_RUNS.last().unwrap().0 > (char::MAX as u32));
+            assert!(SHORT_OFFSET_RUNS.last().unwrap().0 > char::MAX as u32);
             let mut i = 0;
             while i < SHORT_OFFSET_RUNS.len() {
                 assert!(SHORT_OFFSET_RUNS[i].start_index() < OFFSETS.len());
@@ -370,7 +370,7 @@ pub mod cc {
     ];
     pub fn lookup(c: char) -> bool {
         const {
-            assert!(SHORT_OFFSET_RUNS.last().unwrap().0 > (char::MAX as u32));
+            assert!(SHORT_OFFSET_RUNS.last().unwrap().0 > char::MAX as u32);
             let mut i = 0;
             while i < SHORT_OFFSET_RUNS.len() {
                 assert!(SHORT_OFFSET_RUNS[i].start_index() < OFFSETS.len());
@@ -443,7 +443,7 @@ pub mod grapheme_extend {
     #[inline(never)]
     fn lookup_slow(c: char) -> bool {
         const {
-            assert!(SHORT_OFFSET_RUNS.last().unwrap().0 > (char::MAX as u32));
+            assert!(SHORT_OFFSET_RUNS.last().unwrap().0 > char::MAX as u32);
             let mut i = 0;
             while i < SHORT_OFFSET_RUNS.len() {
                 assert!(SHORT_OFFSET_RUNS[i].start_index() < OFFSETS.len());
@@ -605,7 +605,7 @@ pub mod n {
     ];
     pub fn lookup(c: char) -> bool {
         const {
-            assert!(SHORT_OFFSET_RUNS.last().unwrap().0 > (char::MAX as u32));
+            assert!(SHORT_OFFSET_RUNS.last().unwrap().0 > char::MAX as u32);
             let mut i = 0;
             while i < SHORT_OFFSET_RUNS.len() {
                 assert!(SHORT_OFFSET_RUNS[i].start_index() < OFFSETS.len());

--- a/src/tools/unicode-table-generator/src/skiplist.rs
+++ b/src/tools/unicode-table-generator/src/skiplist.rs
@@ -68,10 +68,7 @@ impl RawEmitter {
             assert!(inserted);
         }
 
-        writeln!(
-            &mut self.file,
-            "use super::ShortOffsetRunHeader;\n"
-        ).unwrap();
+        writeln!(&mut self.file, "use super::ShortOffsetRunHeader;\n").unwrap();
         writeln!(
             &mut self.file,
             "static SHORT_OFFSET_RUNS: [ShortOffsetRunHeader; {}] = [{}];",
@@ -111,46 +108,22 @@ impl RawEmitter {
         } else {
             writeln!(&mut self.file, "pub fn lookup(c: char) -> bool {{").unwrap();
         }
-        writeln!(
-            &mut self.file,
-            "    const {{",
-        )
-        .unwrap();
+        writeln!(&mut self.file, "    const {{").unwrap();
         writeln!(
             &mut self.file,
             "        assert!(SHORT_OFFSET_RUNS.last().unwrap().0 > (char::MAX as u32));",
         )
         .unwrap();
-        writeln!(
-            &mut self.file,
-            "        let mut i = 0;",
-        )
-        .unwrap();
-        writeln!(
-            &mut self.file,
-            "        while i < SHORT_OFFSET_RUNS.len() {{",
-        )
-        .unwrap();
+        writeln!(&mut self.file, "        let mut i = 0;").unwrap();
+        writeln!(&mut self.file, "        while i < SHORT_OFFSET_RUNS.len() {{").unwrap();
         writeln!(
             &mut self.file,
             "            assert!(SHORT_OFFSET_RUNS[i].start_index() < OFFSETS.len());",
         )
         .unwrap();
-        writeln!(
-            &mut self.file,
-            "            i += 1;",
-        )
-        .unwrap();
-        writeln!(
-            &mut self.file,
-            "        }}",
-        )
-        .unwrap();
-        writeln!(
-            &mut self.file,
-            "    }}",
-        )
-        .unwrap();
+        writeln!(&mut self.file, "            i += 1;").unwrap();
+        writeln!(&mut self.file, "        }}").unwrap();
+        writeln!(&mut self.file, "    }}").unwrap();
         writeln!(
             &mut self.file,
             "    // SAFETY: We just ensured the last element of `SHORT_OFFSET_RUNS` is greater than `std::char::MAX`",

--- a/src/tools/unicode-table-generator/src/skiplist.rs
+++ b/src/tools/unicode-table-generator/src/skiplist.rs
@@ -111,7 +111,7 @@ impl RawEmitter {
         writeln!(&mut self.file, "    const {{").unwrap();
         writeln!(
             &mut self.file,
-            "        assert!(SHORT_OFFSET_RUNS.last().unwrap().0 > (char::MAX as u32));",
+            "        assert!(SHORT_OFFSET_RUNS.last().unwrap().0 > char::MAX as u32);",
         )
         .unwrap();
         writeln!(&mut self.file, "        let mut i = 0;").unwrap();

